README.md

```diff
@@ -108,24 +119,137 @@ The compiled workflow can be executed via the DNAnexus command line client or we
 
 ![this](doc/bam_chrom_counter.png)
 
-### Strict syntax
+### Validate the workflow
 
 dxCompiler uses [wdlTools](https://github.com/dnanexus/wdlTools), a parser that adheres strictly to the WDL specifications. Most of the problematic automatic type conversions that are allowed by some other WDL runtime engines are not allowed by dxCompiler. Please use the command line tools in wdlTools (e.g. `check` and `lint`) to validate your WDL files before trying to compile them with dxCompiler.
 
 ## CWL
 
-To compile a CWL workflow/tool, you will first need to process it using the following steps:
+### Preprocess CWL workflow
+
+dxCompiler requires the source CWL file to be "packed" as a cwl.json file, which contains a single compound workflow with all the dependent processes included. Additionally, you may need to upgrade the version of your workflow to 1.2. The following examples demonstrate how to do that:
 
-1. Upgrade it to CWL 1.2 if it is not already: 
+1. Install `cwl-upgrader` and upgrade your workflow to CWL 1.2 (if it is not already): 
     ```
-    $ cwl-upgrader my-workflow.cwl
+    $ pip install cwl-upgrader
+    $ cwl-upgrader my-workflow.cwl [subworkflow1.cwl subworkflow2.cwl ...]
     ```
-2. Pack it:
+2. Install `sbpack` package and use the `cwlpack` command on the top-level workflow file to build the "packed" one:
     ```
+    $ pip install https://github.com/rabix/sbpack.git`
     $ cwlpack --add-ids --json my-workflow.cwl > my-workflow.cwl.json
     ```
 3. De-localize all local paths referenced in your CWL: if your CWL specifies a local path, e.g. a schema or a default value for a `file`-type input, you need to upload those files to a DNAnexus project and then replace the local path in your CWL with a DNAnexus URI, e.g. `dx://project-XXX:file-YYY`.
 
+### Compile and run workflow
+The same `bam_chrom_counter` workflow as the example WDL above is now written in CWL v1.0:
+```
+#!/usr/bin/env cwl-runner
+cwlVersion: v1.2
```

Comment:
@commandlinegirl I switched the sample cwl version to 1.2. Was using v1.0 so we can use it as an example for both upgrading and packing.

Reply:
Thanks, I think it's best to allow the user to quickly and easily copy paste the script and compile it right away. We expect the users to write CWL scripts in 1.2 from the start.